enhancements/console/guided-tours.md

```diff
@@ -0,0 +1,144 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+OpenShift's Serverless team has proposed an idea to create a "Guided Tours"
+mechanism which introduces users to various ways of interacting with serverless
+in the Console. Guided Tours should be a mechanism we can use to vastly improve
+our customer's initial user experience on a empty cluster or with all various 
+workflows:
+
+The goal of this proposal is to define a lightweight mechanism for OpenShift's
+Console component, to guide users thought various workflows, and help them
+understand the steps neccesary to get the desired outcome:
+
+* Install operator
+* Deployment of showcase application
+* Cluster settings
+* Networking
+* ...
+
+In order to achieve the desired outcome user needs to create a set of resources,
+that are dependent on each other.
+
+For Guided Tours we need a mechanism for their creation and publishment.
+
+## Motivation
+
+Help users with their understanding the principles of their workflows by
+guiding them though the necessary steps.
+
+### Goals
+
+1. Provide a mechanism to display user guides for various workflows.
+2. Make users understand what are the steps needed to achieve their goals.
+3. Provide API for writing Guided Tours.
+4. Provide new CRD format for writing the Guided Tours.
+5. Have a storage for the Guided Tours.
+
+## Proposal
+
+* Introduce CRD format that will be used for writing Guided Tours.
+* Introduce default storage for Guided Tours.
+  
+### User Stories
+
+#### Story 1
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to install OpenShift Serverless modules (Serving and Eventing) in a cluster.
+
+#### Story 2
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to update an OpenShift cluster.
+
+#### Story 3
+
+As a developer, I need a guide to walk me through how to deploy an existing application as a serverless workload.
+
+#### Story 4
+
+As a operator creator I want to provide operator consumers with a guide on how to install and user the my operator.
+
+### Implementation Details
+
+1. In order to provide mechanism to discribe a Guided Tour, new CRD named `GuidedTours` will be created.
+2. A new `openshift/guided-tours` repository will be created which will contain all supported Guided Tours.
+3. `console-operator` will import all the existing Guided Tours CRs from the `openshift/guided-tours` repository into the `/manifest` directory, so that the CVO can:
+   * create them if the CR doesn't exists.
+   * update the CRs upon the cluster update (since CVO is doing `apply`).
+4. `console-operator` will check for older `GuidedTours` CR versions and remove those that doesn't match the cluster major version, which indicates that new version is not available.
```

Comment:
Do you mean the cluster minor version? (The `y` version in `4.y.z`?)

We might want to support a semver version range here so one tour CR can be used for multiple OpenShift versions.